pkg/blobinfocache/memory/memory.go

```diff
@@ -159,26 +173,50 @@ func (mem *cache) CandidateLocations(transport types.ImageTransport, scope types
 // If !canSubstitute, the returned cadidates will match the submitted digest exactly; if canSubstitute,
 // data from previous RecordDigestUncompressedPair calls is used to also look up variants of the blob which have the same
 // uncompressed digest.
+//
+// If scope is nil, final result also contains blobs with lowest priority without locations that could be possibly be reused.
 func (mem *cache) CandidateLocations2(transport types.ImageTransport, scope types.BICTransportScope, primaryDigest digest.Digest, canSubstitute bool) []blobinfocache.BICReplacementCandidate2 {
-	return mem.candidateLocations(transport, scope, primaryDigest, canSubstitute, true)
+	resAllBlobs := []prioritize.CandidateWithTime{}
+	v2Output := true
+	resultPrioritized := mem.candidateLocations(transport, scope, primaryDigest, canSubstitute, v2Output)
+	mem.mutex.Lock()
+	defer mem.mutex.Unlock()
```

Comment:
Same comments as on the BoltDB side. Why is this not done inside `candidateLocations`?

Is that still WIP?

Reply:
It is moved now, could you PTAL and marked this bit as resolved.